presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcResultSet.java

```diff
@@ -183,26 +185,25 @@ public void testObjectTypes()
 //            ...
 //        });
 
-        checkRepresentation("TIMESTAMP '2018-02-13 13:14:15.227 Europe/Warsaw'", Types.TIMESTAMP /* TODO TIMESTAMP_WITH_TIMEZONE */, (rs, column) -> {
-            assertEquals(rs.getObject(column), Timestamp.valueOf(LocalDateTime.of(2018, 2, 13, 6, 14, 15, 227_000_000))); // TODO this should represent TIMESTAMP '2018-02-13 13:14:15.227 Europe/Warsaw'
+        checkRepresentation("TIMESTAMP '2018-02-13 13:14:15.227 Europe/Warsaw'", Types.TIMESTAMP_WITH_TIMEZONE, (rs, column) -> {
+            assertEquals(rs.getObject(column), ZonedDateTime.of(2018, 2, 13, 13, 14, 15, 227_000_000, ZoneId.of("Europe/Warsaw")));
             assertThrows(() -> rs.getDate(column));
             assertThrows(() -> rs.getTime(column));
-            assertEquals(rs.getTimestamp(column), Timestamp.valueOf(LocalDateTime.of(2018, 2, 13, 6, 14, 15, 227_000_000))); // TODO this should fail, or represent TIMESTAMP '2018-02-13 13:14:15.227'
+            assertThrows(() -> rs.getTimestamp(column)); // this could also return java.sql.Timestamp.valueOf(LocalDateTime.of(2018, 2, 13, 13, 14, 15, 227_000_000))
```

Comment:
We should continue to allow calling `getTimestamp()` to not break applications.

Reply:
Okay, and how do we deal with timestamps that are nor representable as `java.sql.Timestamp` (ie values when JVM zone had forward offset shift)?
just return some near value (as `java.sql.Timestamp.valueOf(LocalDateTime)` probably does)?

Reply:
Yeah, I think using `Timestamp.valueOf(LocalDateTime)` is reasonable here.

That code path goes through the legacy `Timestamp` constructor which takes fields and calls into some `sun.util.calendar` code, so it's not clear what it does, but it seems fine to use the method since it exists in the JDK.